src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,103 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+
+    private T[] elementData;
+    private int size;
+
+    public ArrayList() {
+        elementData = (T[]) new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
 
     @Override
     public void add(T value) {
-
+        if (size + 1 > elementData.length) {
+            ensureCapacity();
+        }
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (size + 1 > elementData.length) {
+            ensureCapacity();
+        }
+        System.arraycopy(elementData, index, elementData, index + 1, size - index);
+        elementData[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index < size) {
+            return elementData[index];
+        } else {
+            throw new ArrayIndexOutOfBoundsException(
+                    "Index [" + index + "] is out of size [" + size + "]");
+        }
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (index >= size) {
+            throw new ArrayIndexOutOfBoundsException(
+                    "Index [" + index + "] is out of size [" + size + "]");
+        }
+        elementData[index] = value;
     }
 
     @Override
     public T remove(int index) {
```

Comment:
let's check index

Reply:
it is checked  inside get() in the first line --> T result = get(index);